gateway-developer-plugin/src/main/java/com/ca/apim/gateway/cagatewayconfig/CAGatewayDeveloper.java

```diff
@@ -92,13 +93,15 @@ public void apply(@NotNull final Project project) {
 
         // add the deployment bundle to the default artifacts
         project.artifacts(artifactHandler -> addBundleArtifact(artifactHandler, packageGW7Task.getBundle(), buildDeploymentBundleTask, project::getName, "deployment"));
-        // the environment bundle is only added after task finish
-        buildEnvironmentBundleTask.doLast(task -> project.artifacts(artifactHandler -> addBundleArtifact(
+        // add the environment bundle to the artifacts only if the environment bundle task was triggered
+        if (project.getGradle().getStartParameter().getTaskNames().contains(BUILD_ENVIRONMENT_BUNDLE)) {
```

Comment:
This change looks simple enough, do we need unit tests? Also what caused the original error?

Reply:
I will look into testing this, the problem is kinda complicated to write I can explain in person if you want :)